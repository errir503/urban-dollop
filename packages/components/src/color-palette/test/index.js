/**
 * External dependencies
 */
import { mount, shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import ColorPalette from '../';

describe( 'ColorPalette', () => {
	const colors = [
		{ name: 'red', color: '#f00' },
		{ name: 'white', color: '#fff' },
		{ name: 'blue', color: '#00f' },
	];
	const currentColor = '#f00';
	const onChange = jest.fn();

	const wrapper = mount(
		<ColorPalette
			colors={ colors }
			value={ currentColor }
			onChange={ onChange }
		/>
	);
	const buttons = wrapper.find( 'Option button' );

	beforeEach( () => {
		onChange.mockClear();
	} );

	test( 'should render a dynamic toolbar of colors', () => {
		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should render three color button options', () => {
		expect( buttons ).toHaveLength( 3 );
	} );

	test( 'should call onClick on an active button with undefined', () => {
		const activeButton = buttons.findWhere( ( button ) =>
			button.hasClass( 'is-pressed' )
		);
		activeButton.simulate( 'click' );

		expect( onChange ).toHaveBeenCalledTimes( 1 );
		expect( onChange ).toHaveBeenCalledWith( undefined );
	} );

	test( 'should call onClick on an inactive button', () => {
		const inactiveButton = buttons
			.findWhere( ( button ) => ! button.hasClass( 'is-pressed' ) )
			.first();
		inactiveButton.simulate( 'click' );

		expect( onChange ).toHaveBeenCalledTimes( 1 );
	} );

	test( 'should call onClick with undefined, when the clearButton onClick is triggered', () => {
		const clearButton = wrapper.find( 'ButtonAction button' );

		expect( clearButton ).toHaveLength( 1 );

		clearButton.simulate( 'click' );

		expect( onChange ).toHaveBeenCalledTimes( 1 );
		expect( onChange ).toHaveBeenCalledWith( undefined );
	} );

	test( 'should allow disabling custom color picker', () => {
		expect(
			shallow(
				<ColorPalette
					colors={ colors }
					disableCustomColors={ true }
					value={ currentColor }
					onChange={ onChange }
				/>
			)
		).toMatchSnapshot();
	} );

	describe( 'Dropdown', () => {
		const dropdown = wrapper.find( 'Dropdown' );

		test( 'should render it correctly', () => {
			expect( dropdown ).toMatchSnapshot();
		} );

		describe( '.renderToggle', () => {
			const isOpen = true;
			const onToggle = jest.fn();

			const renderedToggleButton = mount(
				dropdown.props().renderToggle( { isOpen, onToggle } )
			);

			test( 'should render dropdown content', () => {
				expect( renderedToggleButton ).toMatchSnapshot();
			} );

			test( 'should call onToggle on click.', () => {
				renderedToggleButton.find( 'button' ).simulate( 'click' );

				expect( onToggle ).toHaveBeenCalledTimes( 1 );
			} );
		} );

		describe( '.renderContent', () => {
			const renderedContent = shallow( dropdown.props().renderContent() );

			test( 'should render dropdown content', () => {
				expect( renderedContent ).toMatchSnapshot();
			} );
		} );
	} );
} );
