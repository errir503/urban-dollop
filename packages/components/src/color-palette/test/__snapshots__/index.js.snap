// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPalette Dropdown .renderContent should render dropdown content 1`] = `
<div
  className="components-color-picker is-alpha-disabled"
>
  <div
    className="components-color-picker__saturation"
  >
    <Pure(WithInstanceId(Saturation))
      hsl={
        Object {
          "a": 1,
          "h": 0,
          "l": 50,
          "s": 100,
        }
      }
      hsv={
        Object {
          "a": 1,
          "h": 0,
          "s": 100,
          "v": 100,
        }
      }
      onChange={[Function]}
    />
  </div>
  <div
    className="components-color-picker__body"
  >
    <div
      className="components-color-picker__controls"
    >
      <div
        className="components-color-picker__swatch"
      >
        <div
          className="components-color-picker__active"
          style={
            Object {
              "backgroundColor": "rgb(255, 0, 0)",
            }
          }
        />
      </div>
      <div
        className="components-color-picker__toggles"
      >
        <Pure(WithInstanceId(Hue))
          hsl={
            Object {
              "a": 1,
              "h": 0,
              "l": 50,
              "s": 100,
            }
          }
          onChange={[Function]}
        />
      </div>
    </div>
    <Inputs
      disableAlpha={true}
      hex="#ff0000"
      hsl={
        Object {
          "a": 1,
          "h": 0,
          "l": 50,
          "s": 100,
        }
      }
      onChange={[Function]}
      rgb={
        Object {
          "a": 1,
          "b": 0,
          "g": 0,
          "r": 255,
        }
      }
    />
  </div>
</div>
`;

exports[`ColorPalette Dropdown .renderToggle should render dropdown content 1`] = `
<Fragment>
  <button
    aria-describedby={null}
    aria-expanded={true}
    aria-haspopup="true"
    aria-label="Custom color picker"
    className="components-button is-link"
    onClick={[MockFunction]}
    type="button"
  >
    Custom color
  </button>
</Fragment>
`;

exports[`ColorPalette Dropdown should render it correctly 1`] = `
<Dropdown
  className="components-circular-option-picker__dropdown-link-action"
  contentClassName="components-color-palette__picker"
  renderContent={[Function]}
  renderToggle={[Function]}
>
  <div
    className="components-dropdown components-circular-option-picker__dropdown-link-action"
  >
    <ForwardRef(Button)
      aria-expanded={false}
      aria-haspopup="true"
      aria-label="Custom color picker"
      isLink={true}
      onClick={[Function]}
    >
      <button
        aria-describedby={null}
        aria-expanded={false}
        aria-haspopup="true"
        aria-label="Custom color picker"
        className="components-button is-link"
        onClick={[Function]}
        type="button"
      >
        Custom color
      </button>
    </ForwardRef(Button)>
  </div>
</Dropdown>
`;

exports[`ColorPalette should allow disabling custom color picker 1`] = `
<CircularOptionPicker
  actions={
    <React.Fragment>
      <ButtonAction
        onClick={[Function]}
      >
        Clear
      </ButtonAction>
    </React.Fragment>
  }
  options={
    Array [
      <Option
        aria-label="Color: red"
        isSelected={true}
        onClick={[Function]}
        selectedIconProps={
          Object {
            "fill": "#000000",
          }
        }
        style={
          Object {
            "backgroundColor": "#f00",
            "color": "#f00",
          }
        }
        tooltipText="red"
      />,
      <Option
        aria-label="Color: white"
        isSelected={false}
        onClick={[Function]}
        selectedIconProps={Object {}}
        style={
          Object {
            "backgroundColor": "#fff",
            "color": "#fff",
          }
        }
        tooltipText="white"
      />,
      <Option
        aria-label="Color: blue"
        isSelected={false}
        onClick={[Function]}
        selectedIconProps={Object {}}
        style={
          Object {
            "backgroundColor": "#00f",
            "color": "#00f",
          }
        }
        tooltipText="blue"
      />,
    ]
  }
/>
`;

exports[`ColorPalette should render a dynamic toolbar of colors 1`] = `
<ColorPalette
  colors={
    Array [
      Object {
        "color": "#f00",
        "name": "red",
      },
      Object {
        "color": "#fff",
        "name": "white",
      },
      Object {
        "color": "#00f",
        "name": "blue",
      },
    ]
  }
  onChange={[MockFunction]}
  value="#f00"
>
  <CircularOptionPicker
    actions={
      <React.Fragment>
        <DropdownLinkAction
          buttonProps={
            Object {
              "aria-label": "Custom color picker",
            }
          }
          dropdownProps={
            Object {
              "contentClassName": "components-color-palette__picker",
              "renderContent": [Function],
            }
          }
          linkText="Custom color"
        />
        <ButtonAction
          onClick={[Function]}
        >
          Clear
        </ButtonAction>
      </React.Fragment>
    }
    options={
      Array [
        <Option
          aria-label="Color: red"
          isSelected={true}
          onClick={[Function]}
          selectedIconProps={
            Object {
              "fill": "#000000",
            }
          }
          style={
            Object {
              "backgroundColor": "#f00",
              "color": "#f00",
            }
          }
          tooltipText="red"
        />,
        <Option
          aria-label="Color: white"
          isSelected={false}
          onClick={[Function]}
          selectedIconProps={Object {}}
          style={
            Object {
              "backgroundColor": "#fff",
              "color": "#fff",
            }
          }
          tooltipText="white"
        />,
        <Option
          aria-label="Color: blue"
          isSelected={false}
          onClick={[Function]}
          selectedIconProps={Object {}}
          style={
            Object {
              "backgroundColor": "#00f",
              "color": "#00f",
            }
          }
          tooltipText="blue"
        />,
      ]
    }
  >
    <div
      className="components-circular-option-picker"
    >
      <div
        className="components-circular-option-picker__swatches"
      >
        <Option
          aria-label="Color: red"
          isSelected={true}
          key="#f00"
          onClick={[Function]}
          selectedIconProps={
            Object {
              "fill": "#000000",
            }
          }
          style={
            Object {
              "backgroundColor": "#f00",
              "color": "#f00",
            }
          }
          tooltipText="red"
        >
          <div
            className="components-circular-option-picker__option-wrapper"
          >
            <Tooltip
              text="red"
            >
              <ForwardRef(Button)
                aria-label="Color: red"
                className="components-circular-option-picker__option"
                isPressed={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "#f00",
                    "color": "#f00",
                  }
                }
              >
                <button
                  aria-describedby={null}
                  aria-label="Color: red"
                  aria-pressed={true}
                  className="components-button components-circular-option-picker__option is-pressed"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#f00",
                      "color": "#f00",
                    }
                  }
                  type="button"
                />
              </ForwardRef(Button)>
            </Tooltip>
            <Icon
              fill="#000000"
              icon={
                <SVG
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <Path
                    d="M18.3 5.6L9.9 16.9l-4.6-3.4-.9 1.2 5.8 4.3 9.3-12.6z"
                  />
                </SVG>
              }
            >
              <SVG
                fill="#000000"
                height={24}
                viewBox="0 0 24 24"
                width={24}
                xmlns="http://www.w3.org/2000/svg"
              >
                <svg
                  aria-hidden={true}
                  fill="#000000"
                  focusable={false}
                  height={24}
                  role="img"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <Path
                    d="M18.3 5.6L9.9 16.9l-4.6-3.4-.9 1.2 5.8 4.3 9.3-12.6z"
                  >
                    <path
                      d="M18.3 5.6L9.9 16.9l-4.6-3.4-.9 1.2 5.8 4.3 9.3-12.6z"
                    />
                  </Path>
                </svg>
              </SVG>
            </Icon>
          </div>
        </Option>
        <Option
          aria-label="Color: white"
          isSelected={false}
          key="#fff"
          onClick={[Function]}
          selectedIconProps={Object {}}
          style={
            Object {
              "backgroundColor": "#fff",
              "color": "#fff",
            }
          }
          tooltipText="white"
        >
          <div
            className="components-circular-option-picker__option-wrapper"
          >
            <Tooltip
              text="white"
            >
              <ForwardRef(Button)
                aria-label="Color: white"
                className="components-circular-option-picker__option"
                isPressed={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "color": "#fff",
                  }
                }
              >
                <button
                  aria-describedby={null}
                  aria-label="Color: white"
                  aria-pressed={false}
                  className="components-button components-circular-option-picker__option"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "color": "#fff",
                    }
                  }
                  type="button"
                />
              </ForwardRef(Button)>
            </Tooltip>
          </div>
        </Option>
        <Option
          aria-label="Color: blue"
          isSelected={false}
          key="#00f"
          onClick={[Function]}
          selectedIconProps={Object {}}
          style={
            Object {
              "backgroundColor": "#00f",
              "color": "#00f",
            }
          }
          tooltipText="blue"
        >
          <div
            className="components-circular-option-picker__option-wrapper"
          >
            <Tooltip
              text="blue"
            >
              <ForwardRef(Button)
                aria-label="Color: blue"
                className="components-circular-option-picker__option"
                isPressed={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "#00f",
                    "color": "#00f",
                  }
                }
              >
                <button
                  aria-describedby={null}
                  aria-label="Color: blue"
                  aria-pressed={false}
                  className="components-button components-circular-option-picker__option"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#00f",
                      "color": "#00f",
                    }
                  }
                  type="button"
                />
              </ForwardRef(Button)>
            </Tooltip>
          </div>
        </Option>
      </div>
      <div
        className="components-circular-option-picker__custom-clear-wrapper"
      >
        <DropdownLinkAction
          buttonProps={
            Object {
              "aria-label": "Custom color picker",
            }
          }
          dropdownProps={
            Object {
              "contentClassName": "components-color-palette__picker",
              "renderContent": [Function],
            }
          }
          linkText="Custom color"
        >
          <Dropdown
            className="components-circular-option-picker__dropdown-link-action"
            contentClassName="components-color-palette__picker"
            renderContent={[Function]}
            renderToggle={[Function]}
          >
            <div
              className="components-dropdown components-circular-option-picker__dropdown-link-action"
            >
              <ForwardRef(Button)
                aria-expanded={false}
                aria-haspopup="true"
                aria-label="Custom color picker"
                isLink={true}
                onClick={[Function]}
              >
                <button
                  aria-describedby={null}
                  aria-expanded={false}
                  aria-haspopup="true"
                  aria-label="Custom color picker"
                  className="components-button is-link"
                  onClick={[Function]}
                  type="button"
                >
                  Custom color
                </button>
              </ForwardRef(Button)>
            </div>
          </Dropdown>
        </DropdownLinkAction>
        <ButtonAction
          onClick={[Function]}
        >
          <ForwardRef(Button)
            className="components-circular-option-picker__clear"
            isSecondary={true}
            isSmall={true}
            onClick={[Function]}
          >
            <button
              aria-describedby={null}
              className="components-button components-circular-option-picker__clear is-secondary is-small"
              onClick={[Function]}
              type="button"
            >
              Clear
            </button>
          </ForwardRef(Button)>
        </ButtonAction>
      </div>
    </div>
  </CircularOptionPicker>
</ColorPalette>
`;
